sway/input/seatop_default.c

```diff
@@ -622,13 +634,23 @@ static void handle_rebase(struct sway_seat *seat, uint32_t time_msec) {
 	double sx = 0.0, sy = 0.0;
 	e->previous_node = node_at_coords(seat,
 			cursor->cursor->x, cursor->cursor->y, &surface, &sx, &sy);
-	cursor_do_rebase(cursor, time_msec, e->previous_node, surface, sx, sy);
+
+	if (surface) {
+		if (seat_is_input_allowed(seat, surface)) {
+			wlr_seat_pointer_notify_enter(seat->wlr_seat, surface, sx, sy);
+		}
+	} else {
+		cursor_update_image(cursor, e->previous_node);
+		wlr_seat_pointer_notify_enter(seat->wlr_seat, NULL, 0, 0);
```

Comment:
Isn't this the same as `wlr_seat_pointer_clear_focus`?

Reply:
Hmm I guess the old code had this too. Let's keep it like this in this PR.

Reply:
`wlr_seat_pointer_clear_focus` calls `wlr_seat_pointer_enter`, while this calls `wlr_seat_pointer_notify_enter`. But I agree it's confusing, I had a double-take at this too.